app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,56 @@
+class PostsController < ApplicationController
+  before_action :find_user, only: %i[index show like unlike]
+  before_action :find_post, only: %i[show like unlike]
+
+  def index
+    @posts = @user.posts
+  end
+
+  def show; end
+
+  def new
+    @user = current_user
+    @post = @user.posts.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    if @post.save
+      flash[:notice] = 'Post created successfully.'
+      redirect_to user_path(current_user)
+    else
+      render 'new'
+    end
```

Comment:
* As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=NOT%20use%20rails%20generate%20scaffold%20command.%20The%20point%20of%20this%20project%20is%20to%20show%20you%20how%20to%20set%20up%20controllers%20by%20yourself.) NOT to use rails generate scaffold command. The point of this project is to show you how to set up controllers by yourself. Only create the contorllers and actions that you need For example 👇 
```
# app/controllers/users_controller.rb
class UsersController < ApplicationController
  
  def index; end

  def show; end

end

# app/controllers/posts_controller.rb
class PostsController < ApplicationController
  
  def index; end

  def show; end

end
```